pkg/plugin/pixie/vizier/plugin.go

```diff
@@ -0,0 +1,256 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vizier
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+
+	karmadautil "github.com/karmada-io/karmada/pkg/util"
+	"github.com/sirupsen/logrus"
+	helmclient "helm.sh/helm/v3/pkg/kube"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cli-runtime/pkg/resource"
+
+	"kurator.dev/kurator/pkg/client"
+	"kurator.dev/kurator/pkg/generic"
+	"kurator.dev/kurator/pkg/moreos"
+	"kurator.dev/kurator/pkg/util"
+)
+
+const (
+	crdKind       = "CustomResourceDefinition"
+	vizierCRDName = "viziers.px.dev"
+
+	repoName    = "pixie-operator"
+	repoAddress = "https://pixie-operator-charts.storage.googleapis.com"
+)
+
+var helmBinary = filepath.Join("helm" + moreos.Exe)
+
+type InstallArgs struct {
+	PxNamespace  string
+	CloudAddress string
+	DeployKey    string
+}
+
+type Plugin struct {
+	*client.Client
+
+	args    *InstallArgs
+	options *generic.Options
+	// use helm to install pixie, because of the cli of pixie need login and store token in local and karmada issues:
+	// https://github.com/karmada-io/karmada/issues/2393, https://github.com/karmada-io/karmada/issues/2392
+	helm string
+}
+
+func NewPlugin(s *generic.Options, args *InstallArgs) (*Plugin, error) {
+	plugin := &Plugin{
+		options: s,
+		args:    args,
+	}
+	rest := s.RESTClientGetter()
+	c, err := client.NewClient(rest)
+	if err != nil {
+		return nil, err
+	}
+	plugin.Client = c
+	plugin.helm = "helm"
+
+	return plugin, nil
+}
+
+func (p *Plugin) Execute(cmdArgs, environment []string) error {
+	if err := p.installHelm(); err != nil {
+		return err
+	}
+
+	if err := p.addRepo(); err != nil {
+		return err
+	}
+
+	clusters, err := p.allClusters()
+	if err != nil {
+		return err
+	}
+
+	for _, c := range clusters {
+		clusterClient, err := p.Client.NewClusterClientSet(c)
+		if err != nil {
+			return err
+		}
+
+		if _, err := karmadautil.EnsureNamespaceExist(clusterClient, p.args.PxNamespace, p.options.DryRun); err != nil {
+			return fmt.Errorf("failed to ensure namespace %s, %w", p.args.PxNamespace, err)
+		}
+
+		clusterHelmClient, err := p.Client.NewClusterHelmClient(c)
+		if err != nil {
+			return err
+		}
+
+		_, err = p.applyCrds(clusterHelmClient)
+		if err != nil {
+			return err
+		}
+
+		clusterCRDClient, err := p.Client.NewClusterCRDClientset(c)
+		if err != nil {
+			return err
+		}
+		if err := util.WaitCRDReady(clusterCRDClient, vizierCRDName, p.options.WaitInterval, p.options.WaitTimeout); err != nil {
```

Comment:
Since it create crds before, why cannot you infer the name from `applyCrds`. my concern is this could miss other crds

Reply:
reasonable, updated.